risc0/zkvm/r0vm/r0vm.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 Risc0, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stdio.h>
+
+#include "risc0/zkvm/sdk/cpp/host/receipt.h"
+
+using namespace risc0;
+
+constexpr size_t kMaxStdin = 64 * 1024;
+
+int main(int argc, char* argv[]) {
+  // Check args
+  if (argc != 2) {
+    fprintf(stderr, "Usage: r0vm <elf_file>\n");
+    return 1;
+  }
+  // Consume stdin fully
+  std::vector<char> data(kMaxStdin);
+  size_t count = fread(data.data(), 1, kMaxStdin, stdin);
+  if (count >= kMaxStdin) {
+    fprintf(stderr, "Input too large\n");
+    return 1;
+  }
+  if (!feof(stdin)) {
+    fprintf(stderr, "Error reading from stdin\n");
+    return 1;
+  }
+  try {
+    Prover prover(argv[1]);
+    prover.writeInput(data.data(), count);
+    // Run prover
+    Receipt r = prover.run();
+    // Write private output
+    const auto& out = prover.getOutput();
```

Comment:
Prefer to avoid auto unless it improves readability (e.g. iterator types or templates)